### PR TITLE
feat: Inbox store

### PR DESCRIPTION
**Pull request by muhsin-k**
# Pull Request Template

## Description 

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have tested in both Android and iOS platform
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/utils/camelCase.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import camelcaseKeys from 'camelcase-keys';
+
+export const camelCase = (data: any, options: any) => {
+  return camelcaseKeys(data, options);
+};
```
Comment (pranavrajs) [inline src/utils/camelCase.ts:6]: Is there a specific reason why we’re wrapping it like this here? It doesn’t appear to offer any particular advantage when exporting this.

File: src/store/inbox/inboxSlice.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import { inboxActions } from './inboxActions';
+
+const DEFAULT_INBOX: Readonly<Inbox> = {
+  id: 0,
+  name: 'All Inboxes',
+  channelType: 'Channel::All',
+  avatarUrl: '',
+  channelId: 0,
+  phoneNumber: '',
+  medium: 'Channel::All',
```
Comment (pranavrajs) [inline src/store/inbox/inboxSlice.ts:12]: This doesn't seem like something that should be in the state. I believe this is used to render the option All in the filter. It is a UI logic, shouldn't be part of the global state.

File: src/utils/camelCaseKeys.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import type { Inbox } from '@/types/Inbox';
+import { camelCase } from './camelCase';
+
+export const transformInbox = (inbox: any): Inbox => {
```
Comment (pranavrajs) [inline src/utils/camelCaseKeys.ts:5]: I believe this shouldn’t be a separate helper file; it should be integrated into the apiService. It shouldn’t be used outside of the service.

**Review by pranavrajs — CHANGES_REQUESTED** [review]
Please see the individual comments.

File: src/utils/camelCaseKeys.ts (hunk 3 above)
Comment (muhsin-k) [reply src/utils/camelCaseKeys.ts:5]: @pranavrajs Makes sense. Currently, it's only used in the Inbox store, but we're using the transform methods in other services like Conversation and Message as well. For example, we need to use transformConversation whenever a new conversation is created via action cable event (https://github.com/chatwoot/chatwoot-mobile-app/blob/78b06bd29f4ded4717b39577d6981cb17f442413/src/utils/actionCable.ts#L63). That's why I thought of creating a common utility that we can use anywhere and add the necessary test cases.

**Review by pranavrajs — APPROVED** [review]

**Pull request merged**